### ISSUE TITLE
This issue appears to have been fixed in iOS 6!

### ISSUE THREAD

**Issue by ericf** [scottjehl/iOS-Orientationchange-Fix#16]
Looks like Apple has fixed this issue in Mobile Safari in iOS 6! \o/

The README could be updated to reflect that, along with possibly not applying this hack when iOS 6 is detected.

**Comment by mhulse** [comment]
Just out of curiosity, what's your meta viewport tag look like?

So you're saying this:

```
<meta name="viewport" content="width=device-width, initial-scale=1">
```

... works in the latest iOS? That's good news. Do you happen to know of any blog posts that talk about this fix?

I'll have to test... It would be nice to have the README make note of this.

**Comment by mhulse** [comment]
Just tested in iOS6, and it does appear to be fixed. That's cool! I'm kinda thinking that I'll just let older iOS versions spaz out for the sake of having one less JS file to manage/load.

Not that I don't appreciate this code... It saved my layouts for those older IOSes. :+1: 

Anyone else out there deciding to let the older IOS versions do their thing?

**Comment by kkifa** [comment]
The issue does not appear to be fixed in iOS 6 for me.
Including this javascript did not fix the orientation bug either.

**Comment by scottjehl** [comment]
The code is written to only apply to iOS5 and older, so you can choose to include it or don't - either way, iOS6 won't be affected, as it's finally fixed :)

**Comment by mhulse** [comment]
Anyone have a link to official docs that say it's been fixed? I'd like to spread the word, but I can't find any official documentation.

**Comment by since1976** [comment]
Is it really fixed? I just visited your test page on my iPhone running iOS6 and it is still happening.

**Comment by mhulse** [comment]
I have not been able to find any official Apple docs that say it's been fixed. I'd love to see something official (or, even semi-official).

For those wanting to test for themselves, here's a quick link to the demo page:

[![qr code](http://chart.apis.google.com/chart?cht=qr&chl=http://scottjehl.github.com/iOS-Orientationchange-Fix/without-fix.html&chs=240x240)](http://scottjehl.github.com/iOS-Orientationchange-Fix/without-fix.html)

A quick test on my iPhone4 with iOS6 and it works for me (no bugs).

**Comment by Evanion** [comment]
Well, it works on my iPhone 4S with IOS 6.
But if the user have scaled in portrait mode ... ever so slightly when switching to landscape ... the previous behaviour occurs, and the page zooms in a lot more then relative to the portrait mode.

**Comment by castorda** [comment]
This script does work in iOS5 for me, but not for iOS6. Anyone know why this could be happening? This is what my viewport looks like:
<meta name="viewport" content="width=1021" />

**Comment by jalcide** [comment]
I don't think it's been 100% fixed in iOS6. It seems to still happen, often, on the second rotation. Seemingly, even without being "ever so slightly" scaled.

Here's at least one other who has noticed this, too: http://stackoverflow.com/questions/17678962/mobile-safari-ios6-does-not-change-viewport-width-on-the-second-rotation-wh/18091016#18091016

Any other confirmations on this?

**Comment by amarchen** [comment]
Yep, happens on the second rotation to me too (well, I am the one who posted this question to stackoverflow - http://stackoverflow.com/questions/17678962/mobile-safari-ios6-does-not-change-viewport-width-on-the-second-rotation-wh/18091016#18091016 )
